corehq/apps/analytics/utils.py

```diff
@@ -33,13 +34,28 @@ def is_domain_blocked_from_hubspot(domain):
     ).exists()
 
 
-def hubspot_enabled_for_user(couch_user):
-    if is_email_blocked_from_hubspot(couch_user.username):
+def hubspot_enabled_for_user(user):
+    """
+    Check if a user (or account policy) has given permission for Hubspot
+    analytics to be synced for a given user.
+    :param user: CouchUser or WebUser
+    :return: Boolean (True if hubspot is enabled/allowed)
+    """
+    if is_email_blocked_from_hubspot(user.username):
         return False
-    for domain in couch_user.get_domains():
-        if is_domain_blocked_from_hubspot(domain):
+    if isinstance(user, WebUser):
+        web_user = user
+    else:
+        web_user = WebUser.get_by_username(user.username)
+    if web_user:
+        for domain in web_user.get_domains():
+            if is_domain_blocked_from_hubspot(domain):
```

Comment:
Optional: I see this is not new code, but it seems inefficient to query the database separately for each different domain. The SQL query in `is_domain_blocked_from_hubspot` could be restructured to check for multiple domains with a single query. Could be addressed in a separate PR.

I couldn't easily tell if `hubspot_enabled_for_user` is called in a loop over multiple users, but further db optimizations could be made if that is the case.